transport/grpc/headers.go

```diff
@@ -153,10 +168,17 @@ func metadataToTransportRequest(md metadata.MD) (*transport.Request, error) {
 				request.Encoding = transport.Encoding(getContentSubtype(value))
 			}
 		default:
+			if isReserved(header) || isReservedWithDollarSign(header) {
```

Comment:
can you add a comment on why do we need check both condition?. Same for the `if enforceHeaderRules` condition